src/tools/rados/rados.cc

```diff
@@ -921,9 +924,33 @@ class RadosBencher : public ObjBencher {
     }
 
     if (write_destination & OP_WRITE_DEST_OMAP) {
-      std::map<std::string, librados::bufferlist> omap;
-      omap[string("bench-omap-key-") + stringify(offset)] = bl;
-      op.omap_set(omap);
+      int ret;
+      constexpr uint64_t batch_size = 1000;
+      constexpr uint64_t min_val_len = 4;
+      int batch_count = num_keys/batch_size;
+      auto omap_val_len = std::max(data.op_size/num_keys, min_val_len);
+      bufferlist omap_val;
+      bl.copy(0, omap_val_len, omap_val);
+
+      while (batch_count >= 0) {
+        librados::ObjectWriteOperation op;
+        std::map<std::string, librados::bufferlist> omap;
+        auto omap_size = std::min(batch_size, num_keys);
+        for (unsigned i=0; i < omap_size; i++){
+          std::string omap_key = "bench-omap-key" + stringify(batch_count) + "-" + stringify(i) + stringify(offset);
+          omap[omap_key]=omap_val;
+        }
+        op.omap_set(omap);
+        ret = io_ctx.aio_operate(oid, completions[slot], &op);
```

Comment:
Not sure how this can be addressed in `aio_write` itself other than probably building this support in bencher classes to work with num_keys similar to num_objects. Open to suggestions